src/utils/logger.ts

```diff
@@ -1,20 +1,64 @@
-import Pino, { LoggerOptions } from "pino";
+import { createLogger, format, transports } from "winston";
 import { env } from "./env";
 
-const defaultOptions: LoggerOptions = {
-  redact: ["headers.authorization"],
-  transport: {
-    target: "pino-pretty",
-    options: {
-      translateTime: "HH:MM:ss Z",
-      ignore: "pid,hostname,reqId",
-      singleLine: true,
-    },
-  },
-  level: env.LOG_LEVEL,
+// Custom filter for stdout transport
+const filterOnlyInfoAndWarn = format((info) => {
+  if (info.level === "error") {
+    return false; // Exclude 'error' level logs
+  }
+  return info;
+});
+
+// Custom filter for stderr transport
+const filterOnlyErrors = format((info) => {
+  if (info.level !== "error") {
+    return false; // Exclude non-error level logs
+  }
+  return info;
+});
+
+const colorizeFormat = () => {
+  if (env.NODE_ENV === "development") {
+    return format.colorize();
+  } else {
+    return format.uncolorize();
+  }
 };
 
-const pino = Pino(defaultOptions);
+const winstonLogger = createLogger({
+  level: env.LOG_LEVEL,
+  format: format.combine(
+    format.timestamp(),
+    format.errors({ stack: true }),
+    format.splat(),
+    colorizeFormat(),
+    format.printf(({ timestamp, level, message }) => {
+      return `[${timestamp}] ${level}: ${message}`;
+    }),
+  ),
+  transports: [
+    // Transport for stdout
+    new transports.Console({
+      format: format.combine(
+        filterOnlyInfoAndWarn(),
+        format.printf(({ level, message, timestamp }) => {
+          return `[${timestamp}] ${level}: ${message}`;
```

Comment:
IMO remove timestamp since any logging library should have it as structured metadata. I'm also thinking level can be omitted.

Reply:
We will need this otherwise the logs are not that readable:

I tried `format.simple()`, which prints below:

```
info: [Worker] [processTx] Start {"timestamp":"2024-08-14T04:43:20.876Z"}
```

with `format.json()`:

```
{"level":"\u001b[32minfo\u001b[39m","message":"[Worker] [processTx] Start","timestamp":"2024-08-14T04:44:30.368Z"}
{"level":"\u001b[32minfo\u001b[39m","message":"[Worker] [processTx] Done","timestamp":"2024-08-14T04:44:30.465Z"}
```
_All the above are difficult to read._

-----

but with the `format.printf()` the logs are more readable.

```
[2024-08-14T04:47:15.282Z] info: [Worker] [processTx] Start
[2024-08-14T04:47:15.299Z] info: [Worker] [processTx] Done
```

Will test the branch and check how the logs look on DD.